src/system/sys_profiler.hh

```diff
@@ -376,8 +451,50 @@ protected:
      * Total number of bytes deallocated directly in this frame (not include subframes).
      */
     size_t total_deallocated_;
-
-
+    /**
+     * Maximum number of bytes allocated at one time in this frame (not include subframes).
+     */
+    size_t max_allocated_;
+    /**
+     * Current number of bytes allocated in this frame (not include subframes).
+     */
+    size_t current_allocated_;
+    
+    /**
+     * Number of times new/new[] operator was used in this scope
+     */
+    int alloc_called;
+    /**
+     * Number of times delete/delete[] operator was used in this scope
+     */
+    int dealloc_called;
+    
+    /**
+     * Number of bytes used by Petsc at the start of time-frame
+     */
+    PetscLogDouble petsc_start_memory;
```

Comment:
Make Profiller work also without PETSC,  so use the compiler flag FLOW123D_HAVE_PETSC to switch 
PETSC related code off.  Further even if we have petsc we need a means to turn its ussage off, e.g. in unit tests.
I suggest to add an optional flag to the constructor to turn PETSC memory tracking on.